eis_toolkit/prediction/fuzzy_overlay.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+from beartype import beartype
+
+from eis_toolkit.exceptions import InvalidParameterValueException
+
+
+@beartype
+def and_overlay(data: np.ndarray) -> np.ndarray:
+    """Compute an 'and' overlay operation with fuzzy logic.
+
+    Args:
+        data: The input data as a 3D Numpy array. Each 2D array represents a raster band.
+            Data points should be in the range [0, 1].
+
+    Returns:
+        2D Numpy array with the result of the 'and' overlay operation. Values are in range [0, 1].
+
+    Raises:
+        InvalidParameterValueException: If data values are not in range [0, 1].
+    """
+    if any(band_data.min() < 0 or band_data.max() > 1 for band_data in data):
+        raise InvalidParameterValueException("All data must be in range [0, 1]")
```

Comment:
This check should be refactored to a simple check function so that it can be reused.

```python
def _check_input_data(data):
    if any(band_data.min() < 0 or band_data.max() > 1 for band_data in data):
        raise InvalidParameterValueException("All data must be in range [0, 1]")
```

```suggestion
    _check_input_data(data=data)

```

Reply:
Good point. And the check can be simply `if data.min() < 0 or data.max() > 1:`, no need to loop bands.